src/System.Net.Requests/src/System/Net/RequestStream.cs

```diff
@@ -99,16 +99,52 @@ public override void SetLength(long value)
 
         public override void Write(byte[] buffer, int offset, int count)
         {
+            if (buffer == null)
+            {
+                throw new ArgumentNullException(nameof(buffer));
+            }
+            if (offset < 0 || offset > buffer.Length)
+            {
+                throw new ArgumentOutOfRangeException(nameof(offset));
+            }
+            if (count < 0 || count > (buffer.Length - offset))
+            {
+                throw new ArgumentOutOfRangeException(nameof(count));
+            }
             _buffer.Write(buffer, offset, count);
         }
 
         public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
         {
+            if (buffer == null)
+            {
+                throw new ArgumentNullException(nameof(buffer));
```

Comment:
Basically yes.  There are existing tests we have that can validate that we are returning the exception before the Task instead of in the Task.

See tests like this:

```c#Assert.Throws<ArgumentOutOfRangeException>(() => { Task t = stream.WriteAsync(buffer, 0, buffer.Length + 1); });
```

If the error were returned in the Task, that statement would not throw an exception and the Assert would fail.